service/access.go

```diff
@@ -233,6 +238,13 @@ func (a *accessService) StartAccessUpdater(ctx context.Context) <-chan error {
 	a.tokenCache.StartExpired(ctx, cachePurgePeriod)
 	a.tokenCache.EnableExpiredHook().SetExpiredHook(func(ctx context.Context, k string) {
 		glg.Warnf("the following cache is expired, key: %v", k)
+
+		if val, ok := a.tokenCache.Get(k); ok {
+			if data, ok := val.(*accessCacheData); ok {
+				size := accessCacheMemoryUsage(data)
+				a.memoryUsage -= size
+			}
+		}
```

Comment:
The hook runs the function after the deletion. This may not work...
https://github.com/kpango/gache/blob/f3ad957bd69d82d8fbb38843d691d0d3e42474b7/gache.go#L253-L258

Reply:
Token expired in cache is unsupported use case. Fix by adding warning message.